src/Microsoft.AspNetCore.TestHost/ResponseFeature.cs

```diff
@@ -13,20 +13,51 @@ internal class ResponseFeature : IHttpResponseFeature
     {
         private Func<Task> _responseStartingAsync = () => Task.FromResult(true);
         private Func<Task> _responseCompletedAsync = () => Task.FromResult(true);
+        private HeaderDictionary _headers = new HeaderDictionary();
```

Comment:
Why bother with the explicit backing field for `Headers`? Don't need the concrete type anywhere.

Reply:
Yes you do. IHeaderDictionary.IsReadOnly does not have a setter, only HeaderDictionary.IsReadOnly adds a setter.

Reply:
thx